posydon/unit_tests/utils/test_data_download.py

```diff
@@ -0,0 +1,238 @@
+               "objects without an update on the unit test."
+
+    def test_instance_PATH_TO_POSYDON_DATA(self):
+        assert isinstance(totest.PATH_TO_POSYDON_DATA, (str, bytes,\
+                                                        totest.os.PathLike))
+
+    def test_instance_file(self):
+        assert isinstance(totest.file, (str, bytes, totest.os.PathLike))
+
+    def test_instance_data_url(self):
+        assert isinstance(totest.data_url, (str, bytes, totest.os.PathLike))
+
+    def test_instance_original_md5(self):
+        assert isinstance(totest.original_md5, (str, bytes,\
+                                                totest.os.PathLike))
+
+    def test_instance_ProgressBar(self):
+        assert isclass(totest.ProgressBar)
+
+    def test_instance_data_download(self):
+        assert isroutine(totest.data_download)
+
+
+class TestValues:
+    # check that the values fit
+    def test_value_PATH_TO_POSYDON_DATA(self):
+        assert "POSYDON_data" in totest.PATH_TO_POSYDON_DATA
+
+    def test_value_file(self):
+        assert "POSYDON_data.tar.gz" in totest.file
+
+    def test_value_data_url(self):
+        assert totest.data_url == "https://zenodo.org/record/6655751/files/"+\
+               "POSYDON_data.tar.gz"
+
+    def test_value_original_md5(self):
+        assert totest.original_md5 == "8873544d9a568ebb85bccffbf1bdcd99"
+
+
+class TestFunctions:
+    @fixture
+    def test_path(self, tmp_path):
+        # a temporary path to POSYDON_data for testing
+        return totest.os.path.join(tmp_path, "POSYDON_data")
+
+    @fixture
+    def download_statement(self):
+        # statement that the download started
+        return "Downloading POSYDON data from Zenodo to PATH_TO_POSYDON_DATA="
+
+    @fixture
+    def failed_MD5_statement(self):
+        # statement that MD5 verfication failed
+        return "Failed to read the tar.gz file for MD5 verificaton"
+
+    @fixture
+    def extraction_statement(self):
+        # statement that the tar extraction started
+        return "Extracting POSYDON data from tar file..."
+
+    @fixture
+    def removal_statement(self):
+        # statement that the tar file gets removed
+        return "Removed downloaded tar file."
+
+    # test functions
+    def test_data_download(self, capsys, monkeypatch, test_path,\
+                           download_statement, failed_MD5_statement,\
+                           extraction_statement, removal_statement):
+        def mock_urlretrieve(url, filename=None, reporthook=None, data=None):
+            return None
+        class mock_TarFile:
+            def getmembers(self):
+                return []
+            def extract(self, member, path='', set_attrs=True, *,\
+                        numeric_owner=False, filter=None):
+                return
+        class mock_open:
+            def __init__(self, name=None, mode='r', fileobj=None,\
+                         bufsize=10240, **kwargs):
```

Comment:
what is this "bufsize"? it doesn't show up anywhere else here or  data_download.py

Reply:
We don't care about it currently. This is the definition of a mocked class and it's function, hence I gave it the same arguments are the original function: [tarfile.open()](https://docs.python.org/3/library/tarfile.html#tarfile.open), which it is replacing during the test.